owsi-core/owsi-core-components/owsi-core-component-jpa-more/src/main/java/fr/openwide/core/jpa/more/business/history/service/IAbstractHistoryLogService.java

```diff
@@ -0,0 +1,48 @@
+package fr.openwide.core.jpa.more.business.history.service;
+
+import java.util.Date;
+import java.util.List;
+
+import fr.openwide.core.jpa.business.generic.service.IGenericEntityService;
+import fr.openwide.core.jpa.exception.SecurityServiceException;
+import fr.openwide.core.jpa.exception.ServiceException;
+import fr.openwide.core.jpa.more.business.difference.service.IDifferenceService;
+import fr.openwide.core.jpa.more.business.difference.util.IDifferenceFromReferenceGenerator;
+import fr.openwide.core.jpa.more.business.difference.util.IHistoryDifferenceGenerator;
+import fr.openwide.core.jpa.more.business.history.model.AbstractHistoryLog;
+import fr.openwide.core.jpa.more.business.history.model.AbstractHistoryDifference;
+import fr.openwide.core.jpa.more.business.history.model.bean.AbstractHistoryLogObjectsBean;
+import fr.openwide.core.jpa.more.business.history.util.IDifferenceHandler;
+
+public interface IAbstractHistoryLogService<
```

Comment:
Well, it's *one* of the patterns we use for this sort of things, since we use the term "Generic" for the user services, IIRC. Of course, it's open to interpretation as to whether the *service* is generic, or the *user* is. But so goes for the term "abstract", I think.

I just made a quick search and the only places where this pattern is still used seem to be:

 * IAbstractGenericListItemSearchQuery
 * IAbstractDataUpgradeService
 * And deprecated or soon-to-be-removed interfaces (IAbstractParameterService, IAbstractExecutionDao, IAbstractExecutionService)

Since it's not widely used anymore, maybe it's a good time to switch to another naming scheme for this particular case?